uktrade/data-hub-frontend
# Enforce company contact email uniqueness

**Note** 
Requires API to be deployed soon after this. In the interim it will just mean that there is neither a warning message nor an error when someone attempts to enter a duplicate email for a company. This could be done easily before by ignoring the warning message so it doesn't actually change things functionally.

## Description of change

- Show api form errors on the form (not an error dialogue box)
- Allow the API to validate contact email uniqueness instead of the frontend
- Remove the info box warning that a non-unique email address has been entered and put an error in instead

## Test instructions

To test this fully you will need to run the equivalent branch of api locally (the uniqueness is not enforced on staging yet).

- Go to a company and visit the contacts tab.
- Try and add a second contact with the same email address as another
- An error should show on the form to say that the email should not be duplicated at that company
- Putting in a unique email address should allow the form to submit

## Screenshots
### Before

![image](https://user-images.githubusercontent.com/1234577/160812385-e3a38132-db29-49e3-8f4d-3eb086d08585.png)

### After

![image](https://user-images.githubusercontent.com/1234577/160811917-05c65dbc-e588-44ba-a083-6a1c7dcb4e10.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
